pkg/build/strategies/sti/sti.go

```diff
@@ -272,82 +272,114 @@ func mergeLabels(newLabels, existingLabels map[string]string) map[string]string
+		cmd = path.Join(location, "scripts", api.Run)
 	}
+	return cmd
+}
+
+func (builder *STI) createLabelsForResultingImage() map[string]string {
 	existingLabels, err := builder.docker.GetLabels(builder.config.BuilderImage)
 	if err != nil {
-		glog.Errorf("Unable to read existing labels from current builder image %s", builder.config.BuilderImage)
+		glog.V(0).Infof("error: Unable to read existing labels from current builder image %s", builder.config.BuilderImage)
 	}
 
-	buildImageUser, err := builder.docker.GetImageUser(builder.config.BuilderImage)
-	if err != nil {
-		return err
-	}
+	return mergeLabels(util.GenerateOutputImageLabels(builder.sourceInfo, builder.config), existingLabels)
+}
 
-	resultLabels := mergeLabels(util.GenerateOutputImageLabels(builder.sourceInfo, builder.config), existingLabels)
+func (builder *STI) commitContainer(containerID, cmd, user string, env []string, labels map[string]string) (string, error) {
 	opts := dockerpkg.CommitContainerOptions{
-		Command:     append([]string{}, runCmd),
-		Env:         buildEnv,
+		Command:     []string{cmd},
+		Env:         env,
 		ContainerID: containerID,
 		Repository:  builder.config.Tag,
-		User:        buildImageUser,
-		Labels:      resultLabels,
+		User:        user,
+		Labels:      labels,
 	}
 
 	imageID, err := builder.docker.CommitContainer(opts)
 	if err != nil {
-		return errors.NewCommitError(builder.config.Tag, err)
+		return "", errors.NewCommitError(builder.config.Tag, err)
 	}
 
-	builder.result.Success = true
-	builder.result.ImageID = imageID
+	return imageID, nil
+}
 
-	if len(builder.config.Tag) > 0 {
-		glog.V(1).Infof("Successfully built %s", builder.config.Tag)
-	} else {
-		glog.V(1).Infof("Successfully built %s", imageID)
+func (builder *STI) removePreviousImage() {
+	if !builder.incremental || !builder.config.RemovePreviousImage {
+		return
+	}
+
+	previousImageID, err := builder.docker.GetImageID(builder.config.Tag)
+	if err != nil {
+		glog.V(0).Infof("error: Error retrieving previous image's metadata: %v", err)
+		return
 	}
 
-	if builder.incremental && builder.config.RemovePreviousImage && previousImageID != "" {
+	if previousImageID != "" {
 		glog.V(1).Infof("Removing previously-tagged image %s", previousImageID)
 		if err = builder.docker.RemoveImage(previousImageID); err != nil {
-			glog.Errorf("Unable to remove previous image: %v", err)
+			glog.V(0).Infof("error: Unable to remove previous image: %v", err)
 		}
 	}
+}
 
-	if builder.config.CallbackURL != "" {
-		builder.result.Messages = builder.callbackInvoker.ExecuteCallback(builder.config.CallbackURL,
-			builder.result.Success, resultLabels, builder.result.Messages)
+func (builder *STI) invokeCallbackUrl(resultLabels map[string]string) {
+	if builder.config.CallbackURL == "" {
+		return
 	}
 
-	return nil
+	builder.result.Messages = builder.callbackInvoker.ExecuteCallback(builder.config.CallbackURL,
+		builder.result.Success, resultLabels, builder.result.Messages)
```

Comment:
i guess the general rule is to do what you did to minimize indenting, but in this case it's just more code.